.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,238 @@
+
+## Upstream bugs
+
+In the case the sniff error code doesn't starts with `WPThemeReview`, and instead it starts with `WordPress`, `PHPCompatibility`, or something else, that means that it is and 'upstream' bug coming from either `WPCS`, `PHPCompatibility` or `PHPCS`. You can report the bug here, but the chances are high that you'll be asked to report it in the correct repository instead.
+
+# Contributing patches and new features
+
+## Branches
+
+Ongoing development will be done in the `develop` branch with merges done into `master` once considered stable.
+
+## Picking an open issue
+
+If you want to pick an open issue, please mention that in the issue, and it will be assigned to you. This will prevent the possible double work by different people on the same issue.
+
+## Contributing to the project
+
+If you want to contribute to this project, fork the repo and open a pull request to the `develop` branch. If you have push access to this repo, you can create a feature branch and then open an intra-repo PR from that branch to the `develop`.
+
+# Considerations when writing sniffs
+
+## Unit Testing
+
+All PRs which affect sniffs, whether bug fixes to existing sniffs, or the addition of a new sniff, should be accompanied by unit tests.
+
+### Pre-requisites
+
+* PHP 5.4 or higher
+* WordPress-Coding-Standards 1.0.0 or higher
+* PHP_CodeSniffer 3.3.0 or higher
+* PHPUnit 4.x, 5.x, 6.x or 7.x
+* PHPCompatibility 8.0 or higher
+
+### Composer setup
+
+Use Composer to install all the necessary dependencies to write and run unit tests. Running
+
+```bash
+composer install
+```
+
+from the root of the cloned repository, will install `PHP_CodeSniffer`, `WordPress Coding Standards`, `PHPUnit`, `PHPCompatibility` and `security advisories` which ensures that your application doesn't have installed dependencies with known security vulnerabilities.
+
+### Other setups
+
+If you have PHP_CodeSniffer (PHPCS) and/or WordPress Coding Standards (WordPressCS) already installed, for instance as git clones, because you either contribute to these libraries as well or you want to develop for or test WPThemeReview with bleeding edge versions of either PHP_CodeSniffer or WordPress Coding Standards, you need to take some additional steps to make it all work.
+
+First, make sure you also have PHPCompatibility and make sure the `installed_paths` for PHP_CodeSniffer is setup correctly.
+
+You can see how this can be done by reading the official PHPCS [documentation](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Configuration-Options#setting-the-installed-standard-paths).
+
+#### Method 1
+
+Set up global environment variables to point to PHP_CodeSniffer and WordPress Coding Standards:
+
+```bash
+PHPCS_DIR: I:/path/to/PHP_CodeSniffer/
+WPCS_DIR: I:/path/to/WordPressCS/
```

Comment:
Why is this required? Shouldn't the _dealerdirect/phpcodesniffer-composer-installer_ package take care of this path?

Reply:
Please read the whole document. A composer-based setup is the default, however, for devs, other setups are also possible which allow for testing with, for instance, dev branches of WPCS or PHPCS.
This section explains how to get the unit test working in that case.